### PR TITLE
Ruby 2.4.0-preview2 Released

### PR DESCRIPTION
**Pull request by nurse**

### CODE REVIEW

File: en/news/_posts/2016-09-08-ruby-2-4-0-preview2-released.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+(This is the one usually released as preview1)
+Feel free to
+[send feedback](https://bugs.ruby-lang.org/projects/ruby/wiki/HowToReport)
+since you can still change the features.
+
+## [Unify Fixnum and Bignum into Integer](https://bugs.ruby-lang.org/issues/12005)
+
+Though [ISO/IEC 30170:2012](http://www.iso.org/iso/iso_catalogue/catalogue_tc/catalogue_detail.htm?csnumber=59579)
+doesn't specify details of the Integer class,
+CRuby has two visible Integer classes: Fixnum and Bignum.
```
Comment (unak) [inline en/news/_posts/2016-09-08-ruby-2-4-0-preview2-released.md]: had

File: en/news/_posts/2016-09-08-ruby-2-4-0-preview2-released.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+date: 2016-09-08 18:00:00 +0900
+lang: en
+---
+
+We are pleased to announce the release of Ruby 2.4.0-preview2.
+
+Ruby 2.4.0-preview2 is the second preview of Ruby 2.4.0.
+This preview2 is released earlier than usual because it includes so
+many new features and improvements.
+(This is the one usually released as preview1)
```
Comment (unak) [inline en/news/_posts/2016-09-08-ruby-2-4-0-preview2-released.md]: What do you want to say in this sentence?

File: en/news/_posts/2016-09-08-ruby-2-4-0-preview2-released.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+author: "naruse"
+translator:
+date: 2016-09-08 18:00:00 +0900
+lang: en
+---
+
+We are pleased to announce the release of Ruby 2.4.0-preview2.
+
+Ruby 2.4.0-preview2 is the second preview of Ruby 2.4.0.
+This preview2 is released to get feedback from community.
```
Comment (stomar) [inline en/news/_posts/2016-09-08-ruby-2-4-0-preview2-released.md]: the community

File: en/news/_posts/2016-09-08-ruby-2-4-0-preview2-released.md [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+Feel free to
+[send feedback](https://bugs.ruby-lang.org/projects/ruby/wiki/HowToReport)
+since you can still change the features.
```
Comment (stomar) [inline en/news/_posts/2016-09-08-ruby-2-4-0-preview2-released.md]: s/change/influence/  ???

File: en/news/_posts/2016-09-08-ruby-2-4-0-preview2-released.md [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+
+## [Unify Fixnum and Bignum into Integer](https://bugs.ruby-lang.org/issues/12005)
+
+Though [ISO/IEC 30170:2012](http://www.iso.org/iso/iso_catalogue/catalogue_tc/catalogue_detail.htm?csnumber=59579)
+doesn't specify details of the Integer class,
+CRuby had two visible Integer classes: Fixnum and Bignum.
```
Comment (stomar) [inline en/news/_posts/2016-09-08-ruby-2-4-0-preview2-released.md]: Why use `CRuby` here? Usually we only say `Ruby` on this site, MRI/CRuby is always assumed;
also "CRuby has foo but Ruby 2.4 has bar" is illogical.

Maybe "Prior versions of Ruby had two visible ..." ?

File: en/news/_posts/2016-09-08-ruby-2-4-0-preview2-released.md [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+Ruby 2.4 unifies them into Integer.
+All C extensions which touches Fixnum or Bignum class needs to fix.
```
Comment (stomar) [inline en/news/_posts/2016-09-08-ruby-2-4-0-preview2-released.md]: ... which touch the Fixnum ... need to be fixed.

File: en/news/_posts/2016-09-08-ruby-2-4-0-preview2-released.md [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+
+See also [the ticket](https://bugs.ruby-lang.org/issues/12005) and [akr's slide](http://www.a-k-r.org/pub/2016-09-08-rubykaigi-unified-integer.pdf).
```
Comment (stomar) [inline en/news/_posts/2016-09-08-ruby-2-4-0-preview2-released.md]: slides

File: en/news/_posts/2016-09-08-ruby-2-4-0-preview2-released.md [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+## Debugging
+
+### [Thread#report_on_exception and Thread.report_on_exception](https://bugs.ruby-lang.org/issues/6647)
+
+Ruby ignores exceptions in threads unless another thread explicitly joins them.
+With `report_on_exception = true`,
+you can notice if a thread has died due to an unhandled exception.
+
+Send us feedback what should be the default for `report_on_exception`
+and about report-on-GC.
```
Comment (stomar) [inline en/news/_posts/2016-09-08-ruby-2-4-0-preview2-released.md]: report-on-GC hasn't been mentioned previously, does it need some explanation?

(I do not know what is meant but I am not very much into Ruby development)

File: en/news/_posts/2016-09-08-ruby-2-4-0-preview2-released.md [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+include enough information for debugging.
+Ruby 2.4's deadlock detection shows threads with their backtrace and
+dependent threads.
+
+Try and enjoy programming with Ruby 2.4.0-preview2, and
+[send us feedback](https://bugs.ruby-lang.org/projects/ruby/wiki/HowToReport)!
+
+## Notable Changes since 2.3
+
+* Support OpenSSL 1.1.0
```
Comment (stomar) [inline en/news/_posts/2016-09-08-ruby-2-4-0-preview2-released.md]: will the above mentioned changes be included, at least for the final release post?

Comment (nurse) [inline en/news/_posts/2016-09-08-ruby-2-4-0-preview2-released.md]: Yes, because OpenSSL 1.1.0 is drastically incompatible with 1.x API.

Comment (stomar) [inline en/news/_posts/2016-09-08-ruby-2-4-0-preview2-released.md]: Sorry, I meant will the changes that were already explained above (like Integer, ...) also appear on this list?

If not, the section title should be "**Other** Notable Changes".

**Comment by stomar** [comment]
I took the liberty to apply other small fixes in 910d06106df5f83e77126f0bf10fb3005b0062bd.

File: en/news/_posts/2016-09-08-ruby-2-4-0-preview2-released.md (hunk 9 above)
Comment (nurse) [inline en/news/_posts/2016-09-08-ruby-2-4-0-preview2-released.md]: Ah, it should be "Other Notable Changes".

Comment (stomar) [inline en/news/_posts/2016-09-08-ruby-2-4-0-preview2-released.md]: Fixed with 53f1a0b9f62f3248f064f032e54e575c4e94c725.

Comment (nurse) [inline en/news/_posts/2016-09-08-ruby-2-4-0-preview2-released.md]: thanks!

**Pull request merged**